src/imagecapturecore.cs

```diff
@@ -0,0 +1,900 @@
+
+		[Export ("device:didEncounterError:")]
+		void DidEncounterError (ICDevice device, [NullAllowed] NSError error);
+
+		[Export ("device:didReceiveButtonPress:")]
+		void DidReceiveButtonPress (ICDevice device, NSString /* ICButtonType */ buttonType);
+
+		[Export ("device:didReceiveCustomNotification:data:")]
+		void DidReceiveCustomNotification (ICDevice device, NSDictionary<NSString, NSObject> notification, NSData data);
+	}
+
+	[BaseType (typeof(NSObject))]
+	interface ICDevice {
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		IICDeviceDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Assign)]
+		NSObject WeakDelegate { get; set; }
+
+		[Export ("type")]
+		ICDeviceType Type { get; }
+
+		[NullAllowed, Export ("name")]
+		string Name { get; }
+
+		[NullAllowed, Export ("icon")]
+		CGImage Icon { get; }
+
+		[Export ("capabilities")]
+		string[] Capabilities { get; }
+
+		[Export ("modulePath")]
+		string ModulePath { get; }
+
+		[Export ("moduleVersion")]
+		string ModuleVersion { get; }
+
+		[Export ("moduleExecutableArchitecture")]
+		int ModuleExecutableArchitecture { get; }
+
+		[Export ("remote")]
+		bool Remote { [Bind ("isRemote")] get; }
+
+		[Deprecated (PlatformName.MacOSX, 10, 13)]
+		[Export ("shared")]
+		bool Shared { [Bind ("isShared")] get; }
+
+		[Deprecated (PlatformName.MacOSX, 10, 13)]
+		[Export ("hasConfigurableWiFiInterface")]
+		bool HasConfigurableWiFiInterface { get; }
+
+		[BindAs (typeof (ICTransportType))]
+		[Export ("transportType")]
+		NSString TransportType { get; }
+
+		[Export ("usbLocationID")]
+		int UsbLocationId { get; }
+
+		[Export ("usbProductID")]
+		int UsbProductId { get; }
+
+		[Export ("usbVendorID")]
+		int UsbVendorId { get; }
+
+		[Export ("fwGUID")]
+		long FwGuid { get; }
+
+		[NullAllowed, Export ("serialNumberString")]
+		string SerialNumberString { get; }
+
+		[NullAllowed, Export ("locationDescription")]
+		string LocationDescription { get; }
+
+		[Export ("hasOpenSession")]
+		bool HasOpenSession { get; }
+
+		[NullAllowed, Export ("UUIDString")]
+		string UuidString { get; }
+
+		[NullAllowed, Export ("persistentIDString")]
+		string PersistentIdString { get; }
+
+		[Export ("buttonPressed")]
+		string ButtonPressed { get; }
+
+		[NullAllowed, Export ("autolaunchApplicationPath")]
+		string AutolaunchApplicationPath { get; set; }
+
+		[NullAllowed, Export ("userData")]
+		NSMutableDictionary UserData { get; }
+
+		[Export ("requestOpenSession")]
+		void RequestOpenSession ();
+
+		[Export ("requestCloseSession")]
+		void RequestCloseSession ();
+
+		[Export ("requestYield")]
+		void RequestYield ();
+
+		[Export ("requestSendMessage:outData:maxReturnedDataSize:sendMessageDelegate:didSendMessageSelector:contextInfo:")]
+		unsafe void RequestSendMessage (uint messageCode, NSData data, uint maxReturnedDataSize, NSObject sendMessageDelegate, Selector selector, [NullAllowed] IntPtr contextInfo);
```

Comment:
```suggestion
	void RequestSendMessage (uint messageCode, NSData data, uint maxReturnedDataSize, NSObject sendMessageDelegate, Selector selector, [NullAllowed] IntPtr contextInfo);
```

`unsafe` is not needed.